examples/vnet/doca_gw.c

```diff
@@ -159,21 +199,47 @@ int doca_gw_port_stop(struct doca_gw_port *port)
     return 0;
 }
 
-
 struct doca_gw_pipeline *doca_gw_create_pipe(struct doca_gw_pipeline_cfg *cfg, struct doca_gw_error *err)
 {
     static uint32_t pipe_id = 1;
-    struct doca_gw_pipeline *pl = malloc(sizeof(struct doca_gw_pipeline));
-    memset(pl,0,sizeof(struct doca_gw_pipeline));
+    struct doca_gw_pipeline *pl;
 
-    if (cfg != NULL && pl != NULL) {
-        // allocate what is needed
-        pl->id = pipe_id++;
-    }
+	pl = malloc(sizeof(struct doca_gw_pipeline));
+    if (cfg == NULL || pl == NULL)
+		return NULL;
+    memset(pl,0,sizeof(struct doca_gw_pipeline));
+	LIST_INIT(&pl->entry_list);
```

Comment:
This should be in dpdk layer, because you create it threre and you have the port, so add the entry to port on dpdk layer.

Reply:
it's need more code to do it ,  can commit  with another commit ?